tests/functional/snowflake_view_dependency/test_snowflake_view_dependency.py

```diff
@@ -0,0 +1,127 @@
+import pytest
+from dbt.tests.util import run_dbt, check_relations_equal
+
+_MODELS__DEPENDENT_MODEL_SQL = """
+
+{% if var('dependent_type', 'view') == 'view' %}
+    {{ config(materialized='view') }}
+{% else %}
+    {{ config(materialized='table') }}
+{% endif %}
+
+select * from {{ ref('base_table') }}
+
+"""
+
+
+
+_MODELS__BASE_TABLE_SQL = """
+
+{{ config(materialized='table') }}
+
+select *
+    {% if var('add_table_field', False) %}
+        , 1 as new_field
+    {% endif %}
+
+from {{ ref('people') }}
+
+"""
+
+_SEEDS__PEOPLE_CSV = """id,name
+1,Drew
+2,Jake
+3,Connor
+"""
+
+class TestSnowflakeLateBindingViewDependency:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "dependent_model.sql":  _MODELS__DEPENDENT_MODEL_SQL,
+            "base_table.sql": _MODELS__BASE_TABLE_SQL
+        }
+
+    @pytest.fixture(scope="class")
+    def seeds(self):
+        return {"people.csv": _SEEDS__PEOPLE_CSV}
+
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            'config-version': 2,
+            'seed-paths': ['seeds'],
+            'seeds': {
+                'quote_columns': False,
+            },
+            'quoting': {
+                'schema': False,
+                'identifier': False
+            }
+        }
+
+    """
+    Snowflake views are not bound to the relations they select from. A Snowflake view
+    can have entirely invalid SQL if, for example, the table it selects from is dropped
+    and recreated with a different schema. In these scenarios, Snowflake will raise an error if:
+    1) The view is queried
+    2) The view is altered
+
+    dbt's logic should avoid running these types of queries against views in situations
+    where they _may_ have invalid definitions. These tests assert that views are handled
+    correctly in various different scenarios
+    """
+
+    def test__snowflake__changed_table_schema_for_downstream_view(self, project):
+        results = run_dbt(["seed"])
+        assert len(results) == 1
+
+        results = run_dbt(["run"])
+        assert len(results) == 2
+        check_relations_equal(project.adapter, ["PEOPLE", "BASE_TABLE"])
+        check_relations_equal(project.adapter, ["PEOPLE", "DEPENDENT_MODEL"])
```

Comment:
Lines 77-83 look like 97-103. If this is always the setup for a test in this test class, we could bundle that in a startup method so that the contents of the test case are specific to the functionality being tested.

Reply:
created a method